Cannot build Godot with most modules disabled - error in linking step

### Godot version

v4.2.beta.custom_build. e8d57afae

### System information

Ubuntu 22.04 CI

### Issue description

```
 scons -j2 dev_build=yes target=editor use_asan=yes use_ubsan=yes linker=gold module_basis_universal_enabled=no module_bmp_enabled=no module_camera_enabled=no module_csg_enabled=no module_cvtt_enabled=no module_dds_enabled=no module_denoise_enabled=no module_enet_enabled=no module_etcpak_enabled=no brotli=no module_gltf_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_jpg_enabled=no module_lightmapper_rd_enabled=no module_mbedtls_enabled=no module_meshoptimizer_enabled=no module_minimp3_enabled=no module_mobile_vr_enabled=no module_msdfgen_enabled=no module_multiplayer_enabled=no module_noise_enabled=no module_ogg_enabled=no module_openxr_enabled=no module_raycast_enabled=no module_regex_enabled=no module_squish_enabled=no graphite=no module_text_server_adv_enabled=no module_tga_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_upnp_enabled=no module_vhacd_enabled=no module_vorbis_enabled=no module_webp_enabled=no module_webrtc_enabled=no module_websocket_enabled=no module_webxr_enabled=no module_xatlas_unwrap_enabled=no
```

```
Linking Static Library core/libcore.linuxbsd.editor.dev.x86_64.san.a ...
Ranlib Library core/libcore.linuxbsd.editor.dev.x86_64.san.a ...
Linking Program bin/godot.linuxbsd.editor.dev.x86_64.san ...
main/main.cpp:1557: error: undefined reference to 'GDScriptLanguageServer::port_override'
main/main.cpp:1557: error: undefined reference to 'GDScriptLanguageServer::port_override'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.editor.dev.x86_64.san] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:44:31.753]
````

### Steps to reproduce

Above

### Minimal reproduction project

Above

--- a/modules/gdscript/SCsub
+++ b/modules/gdscript/SCsub
@@ -19,6 +19,8 @@ if env.editor_build:
         # Using a define in the disabled case, to avoid having an extra define
         # in regular builds where all modules are enabled.
         env_gdscript.Append(CPPDEFINES=["GDSCRIPT_NO_LSP"])
+        # Also needed in main env to unexpose --lsp-port option.
+        env.Append(CPPDEFINES=["GDSCRIPT_NO_LSP"])
 
 
 if env["tests"]:

